l10n_it_fatturapa_in_pec/models/mail.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Sergio Corato (https://efatto.it)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+import re
+import zipfile
+import io
+import os
+import base64
+from odoo import api, models
+from odoo.tools import config
+
+_logger = logging.getLogger(__name__)
+
+FATTURAPA_IN_REGEX = "^[A-Z]{2}[a-zA-Z0-9]{11,16}_[a-zA-Z0-9]{,5}.(xml|zip)"
```

Comment:
Sarebbe meglio non ripetere queste variabili, così come il check
```
any("@pec.fatturapa.it" in x for x in [
            message.get('Reply-To', ''),
            message.get('From', ''),
            message.get('Return-Path', '')
        ])
```

In generale le dipendenze di questi 3 moduli mi sembrano un po' intricate (ad esempio `fatturapa_in_pec` dipende da `fatturapa_out_pec`), io farei una struttura come segue:

- `l10n_it_sdi_channel` dipende da `account` e `fetchmail`
- `l10n_it_fatturapa_pec` (da creare, contenente le variabili comuni tipo le varie`REGEX`, il check detto sopra e altre cose utili ad entrambi i moduli) dipende da `l10n_it_fatturapa` e `l10n_it_sdi_channel`
- `l10n_it_fatturapa_in_pec` dipende da `l10n_it_fatturapa_in` e `l10n_it_fatturapa_pec`
- `l10n_it_fatturapa_out_pec` dipende da `l10n_it_fatturapa_out` e `l10n_it_fatturapa_pec`

Credo che le dipendenze siano corrette ma se sbaglio mi corrigerete!

Reply:
@SimoRubi @sergiocorato oppure fondere insieme  `l10n_it_fatturapa_in_pec`  e `l10n_it_fatturapa_out_pec` in un unico modulo `l10n_it_fatturapa_pec`?
Non mi porrei il problema di permettere di gestire solo l'invio o solo la ricezione

Reply:
@eLBati @SimoRubi assolutamente d'accordo

Reply:
direi che l10n_it_sdi_channel dipende da l10n_it_fatturapa per andare ad inserire i campi nella vista configurazione vicino a quelli inseriti da fatturapa

Reply:
Se ho capito bene lo scopo del model `sdi.channel`, il modulo `l10n_it_sdi_channel` potrebbe in teoria dipendere solo da `fetchmail` e si potrebbero spostare le varie cose che lo legano ad `account` nel nuovo `l10n_it_fatturapa_pec` suggerito da @eLBati.

Reply:
Il sistema di interscambio esiste solo per la fatturazione elettronica. Non penso sia utile astrarlo dai moduli di fatturazione. Preparo le modifiche nella PR https://github.com/efatto/l10n-italy/pull/2

Reply:
@eLBati mergiata

Reply:
@sergiocorato vedo conflitti con 10.0
Provo a fare rebase.
Poi per te va bene se faccio push direttamente sul tuo branch `10.0-imp_fatturapa_in_pec`?